pkg/scalers/azure_pipelines_scaler.go

```diff
@@ -306,13 +345,63 @@ func validatePoolID(ctx context.Context, poolID string, metadata *azurePipelines
 	return result.ID, nil
 }
 
-func getAzurePipelineRequest(ctx context.Context, urlString string, metadata *azurePipelinesMetadata, httpClient *http.Client) ([]byte, error) {
+func getToken(ctx context.Context, logger logr.Logger, metadata *azurePipelinesMetadata, podIdentity kedav1alpha1.AuthPodIdentity, scope string) (string, error) {
+	cred, err := azure.NewChainedCredential(logger, podIdentity.GetIdentityID(), podIdentity.Provider)
+	if err != nil {
+		return "", err
+	}
```

Comment:
We have to cache this `cred` var because it has state (basically it knows which is the used authentication mechanism used to refresh the token using it again instead of checking all of them again)